Journalist can categorize Articles depending on topic

AlexSaxena opened the pull request:
[PT-link](https://www.pivotaltracker.com/story/show/169253782)

## PR content
In this PR we are adding a feature where a journalist can choose a category for given article.

## This PR contains:
- Unit tests for model and factories
- Request spec tests for including categories
- Seeded Categories

## What have we learned?
- Improved understanding of Rails and Associations 
- Improved understanding of RSpec and related technologies

AlexSaxena committed:
Adds categories to articles table.

AlexSaxena committed:
Renames the table categories to category,
Adds associations between article and category.

Commit:
Merge branch 'categories' of github.com:AlexSaxena/fake_news_backend into categories

Commit:
Generate Category model & migration file

Migrate foreign key related to article.

Commit:
Add validations for category model

AlexSaxena committed:
Add has_many association for articles & categories
Generates a categories controller.

Commit:
Replace category column with foreign key

Commit:
Replace old syntax to fix specs

AlexSaxena committed:
Refactors Spec_file tests to match new Categories

Commit:
Add category variable to Articles controller

Pass category as an argument in 'Article.create' method.
Remove comments.

AlexSaxena committed:
Adds Subsriber role to Seed file;
Adds Categories to seed file,
Adds Articles to seed file.

Commit:
Accept incoming changes

AlexSaxena committed:
Merge branch 'development' of https://github.com/CraftAcademy/fake_news_backend into categories
Merges latest upstream to branch.

Commit:
Merge branch 'categories' of github.com:AlexSaxena/fake_news_backend into categories

Accept incoming changes from upstream.

AlexSaxena committed:
Removes out of scope file.

AlexSaxena committed:
Refactors code.

clalil commented on app/controllers/v1/articles_controller.rb:
```diff
@@ -21,8 +21,13 @@ def show
 
   def create
+    
+
```
Are these spaces in the actual file as well or is it just Github magic?

Commit:
Remove seeded articles (moved to different branch)

Commit:
Re-add categories and articles to seed file

faraznaeem commented on spec/models/category_spec.rb:
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
```
no test for the association's

GergKllai1 replied:
Still no tests for assications

Commit:
Add association check to Article model spec

Add check for factory to Category model spec.
Fix error in Article factory due to naming different convention for
images.
Add images (from xkcd) for seeding.

Commit:
Increase number of articles in seed file

GergKllai1 commented on spec/requests/v1/api_provides_possibility_for_journalist_to_create_article_spec.rb:
```diff
@@ -1,5 +1,6 @@
 RSpec.describe 'Can create article with attributes' do
   let(:journalist) { create(:user, role: 'journalist') }
+  let(:category) {FactoryBot.create(:category)}
```
You don't need FactoryBot, remove it for consistency

GergKllai1 commented on db/seeds.rb:
```diff
@@ -1,2 +1,72 @@
-users = User.create(email: 'user@mail.com', password: 'password')
+User.destroy_all
+
+user = User.create(email: 'user@mail.com', encrypted_password: 'password', role: 'user')
```
it is just password, for consistency change it pls

AlexSaxena committed:
Adds test for the association's,
Refactors code in Seed file and spec file.

AlexSaxena committed:
Refactors code in category spec
